corehq/apps/settings/forms.py

```diff
@@ -54,8 +56,19 @@ def __init__(self, user, *args, **kwargs):
             ),
         )
 
-    def clean_new_password1(self):
-        return clean_password(self.cleaned_data.get('new_password1'))
+    def clean_old_password(self):
+        from corehq.apps.hqwebapp.utils import decode_password
+        old_password = decode_password(self.cleaned_data['old_password'])
+        if old_password == '':
+            raise ValidationError(
+                _("Password cannot be empty"), code='password_empty',
+            )
+        if not self.user.check_password(old_password):
+            raise forms.ValidationError(
+                self.error_messages['password_incorrect'],
+                code='password_incorrect',
+            )
```

Comment:
It's surprising to me that we need to raise these two validation errors in this PR since they should be handled at other places in the code.

Could it be changed to something like (a little tricky, but try/finally is necessary if `clean_old_password` is called multiple times per request):
```py
def clean_old_password(self):
    encoded_pass = self.cleaned_data['old_password']
    cleaned_pass = decode_password(encoded_pass)
    self.cleaned_data['old_password'] = cleaned_pass
    try:
        return super(HQPasswordChangeForm, self).clean_old_password()
    finally:
        self.cleaned_data['old_password'] = encoded_pass
```

If only called once per request:
```py
def clean_old_password(self):
    self.cleaned_data['old_password'] = decode_password(
        self.cleaned_data['old_password'])
    return super(HQPasswordChangeForm, self).clean_old_password()
```